extractor/src/main/java/org/schabi/newpipe/extractor/utils/AbbreviationHashMap.java

```diff
@@ -0,0 +1,156 @@
+package org.schabi.newpipe.extractor.utils;
+
+import java.util.HashMap;
+
+
+/**
+ * Map matching abbreviations with their English equivalents
+ *
+ * @author B0pol
+ * <p>
+ * By using this map, you can replace the abbreviations used for numbers in the 80 languages supported by YouTube
+ * With their English equivalent.
+ * </p>
+ * <p>
+ * Some language use more abbreviations for numbers: east-asian languages have abbreviations for ten thousand,
+ * and hundred million, indo-arabic languages have abbreviations for a hundred thousand and ten million,
+ * then we replace ten thousand by {@link #tenThousandAbbreviation},
+ * hundred thousand by {@link #hundredThousandAbbreviation},
+ * ten million by {@link #tenMillionAbbreviation},
+ * hundred million by {@link #hundredMillionAbbreviation}.
+ * </p>
+ * <p>
+ * The languages using the abbreviation is commented with the language code at the left.
+ * @see <a href="https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes">Wikipedia page of language codes</a>
+ * </p>
+ */
+public class AbbreviationHashMap {
```

Comment:
I don't really understand. You want me to gather data from Unicode, instead of tests?
So I would have to check Unicode, find if it's not already in the map, add them if, and do that 80 times?
It would be faster to find channels at each range, run tests, see failed one, if we get error rerun / add in the map.

Reply:
I don't understand. You want me to check unicode, check if something is missing, and add it to the map, and **do that 80 times?**
At this point, it would be faster to run tests with more channels, and I will do it (until you explain me what you mean by the other approach.